vcd/resource_vcd_org_group.go

```diff
@@ -0,0 +1,192 @@
+		Read:   resourceVcdOrgGroupRead,
+		Update: resourceVcdOrgGroupUpdate,
+		Delete: resourceVcdOrgGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdOrgGroupImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true, // vCD does not allow to change group name
+				Description: "SAML group name",
+			},
+			"provider_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true, // vCD does not allow to change provider type
+				Description:  "SAML group name",
+				ValidateFunc: validation.StringInSlice([]string{"SAML", "INTEGRATED"}, false),
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description",
+			},
+			"role": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Existing role name to assign",
+			},
+		},
+	}
+}
+
+func resourceVcdOrgGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	roleName := d.Get("role").(string)
+	role, err := adminOrg.GetRoleReference(roleName)
+	if err != nil {
+		return fmt.Errorf("unable to find role %s: %s", roleName, err)
+	}
+
+	newGroup := govcd.NewGroup(&vcdClient.Client, adminOrg)
+	groupDefinition := types.Group{
+		Name:         d.Get("name").(string),
+		Role:         role,
+		ProviderType: d.Get("provider_type").(string),
+		Description:  d.Get("description").(string),
+	}
+	newGroup.Group = &groupDefinition
+
+	createdGroup, err := adminOrg.CreateGroup(newGroup.Group)
+	if err != nil {
+		return fmt.Errorf("error creating group %s: %s", groupDefinition.Name, err)
+	}
+
+	d.SetId(createdGroup.Group.ID)
+
+	return resourceVcdOrgGroupRead(d, meta)
+}
+
+func resourceVcdOrgGroupRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	group, err := adminOrg.GetGroupById(d.Id(), false)
+	if govcd.IsNotFound(err) {
+		log.Printf("error finding group for deletion %s: %s. Removing from state", d.Id(), err)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error finding group for deletion  %s: %s", d.Id(), err)
+	}
+
+	d.Set("name", group.Group.Name)
+	d.Set("description", group.Group.Description)
+	d.Set("role", group.Group.Role.Name)
+	d.Set("provider_type", group.Group.ProviderType)
```

Comment:
Input can be lower case and upper case, but will be return only in `one` case.

Reply:
Nope. There is validation on the field:
```go
ValidateFunc: validation.StringInSlice([]string{"SAML", "INTEGRATED"}, false),
```

Reply:
false/ true - without code hard to guess which is which :)